common/selectors/selectAndDedupe.js

```diff
@@ -14,7 +16,21 @@ dedupe.defaults.createKey = site => site.cache_key || site.hostname || site.url;
 module.exports = function selectAndDedupe(group) {
   return group.reduce((result, options, index, arr) => {
     let current;
-    let sites = options.defaults ? options.sites.concat(options.defaults) : options.sites;
+    let sites = options.sites;
+
+    if (options.defaults) {
+      // Dedupe the defaults first.
+      let defaults = dedupe.group([sites, options.defaults])[1];
+      // Fill in any empty slots with default sites.
+      sites.forEach((site, i) => {
+        if (!site) {
+          sites[i] = defaults.shift();
```

Comment:
Any chance that `defaults.shift()` could return an `undefined`, i.e. running out of the `defaults`? Imaging a user has pinned two links at slot 1 and 12 in the "showMore" experiment, then he clears the whole history. In that case the default links (as currently 6 tiles) won't be enough to populate those 10 slots.